src/utils/dockerContextManager.ts

```diff
@@ -9,13 +9,21 @@ import * as fs from 'fs';
 import * as url from 'url';
 import { workspace, WorkspaceConfiguration } from 'vscode';
 import { parseError } from "vscode-azureextensionui";
+import LineSplitter from '../debugging/coreclr/lineSplitter';
 import { ext } from '../extensionVariables';
 import { localize } from '../localize';
 import { LocalOSProvider } from './LocalOSProvider';
-import { execAsync } from './spawnAsync';
+import { execAsync, spawnAsync } from './spawnAsync';
 import { timeUtils } from './timeUtils';
 
-const ContextInspectExecOptions: ExecOptions = { timeout: 5000 }
+// CONSIDER
+// Any of the commands related to Docker context can take a very long time to execute (a minute or longer)
+// if the current context refers to a remote Docker engine that is unreachable (e.g. machine is shut down).
+// Consider having our own timeout for execution of any context-related Docker CLI commands.
+// The following timeout is for _starting_ the command only; in the current implementation there is no timeot
```

Comment:
Right. For `refreshDockerode` we had to do a `Promise.race` to make a _real_ timeout. It might be helpful for `spawnAsync` to have an option that provides real timeout.

Reply:
<You can resolve this, just putting my thoughts into writing>

Reply:
Yep, I saw that.